### PR TITLE
Fix clusterName: null issue

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

File: test/client_reset_test.dart [hunk 1]
```diff
@@ -393,33 +393,33 @@ Future<void> main([List<String>? args]) async {
...
 
   baasTest('onManualResetFallback is reported after async onAfterReset throws', (appConfig) async {
     final app = App(appConfig);
     final user = await getIntegrationUser(app);
-    int onBeforeResetOccured = 0;
-    int onAfterResetOccured = 0;
-    int manualResetFallbacKOccured = 0;
+    int onBeforeResetOccurred = 0;
+    int onAfterResetOccurred = 0;
+    int manualResetFallbacKOccurred = 0;
```
Comment (desistefanova) [inline test/client_reset_test.dart:405]:
```suggestion
    int manualResetFallbackOccurred = 0;
```

File: test/client_reset_test.dart [hunk 2]
```diff
@@ -393,33 +393,33 @@ Future<void> main([List<String>? args]) async {
...
         onAfterReset: (beforeResetRealm, afterResetRealm) async {
           await Future<void>.delayed(Duration(seconds: 1));
-          onAfterResetOccured++;
+          onAfterResetOccurred++;
           throw Exception("Cause onManualResetFallback");
         },
         onManualResetFallback: (clientResetError) {
-          manualResetFallbacKOccured++;
-          if (onAfterResetOccured == 0) {
+          manualResetFallbacKOccurred++;
```
Comment (desistefanova) [inline test/client_reset_test.dart:421]:
```suggestion
          manualResetFallbackOccurred++;
```

File: test/client_reset_test.dart [hunk 3]
```diff
@@ -433,9 +433,9 @@ Future<void> main([List<String>? args]) async {
 
     await waitFutureWithTimeout(manualResetFallbacKCompleter.future, timeoutError: "onManualResetFallback is not reported.");
 
-    expect(manualResetFallbacKOccured, 1);
-    expect(onAfterResetOccured, 1);
-    expect(onBeforeResetOccured, 1);
+    expect(manualResetFallbacKOccurred, 1);
```
Comment (desistefanova) [inline test/client_reset_test.dart:436]:
```suggestion
    expect(manualResetFallbackOccurred, 1);
```

File: lib/src/cli/atlas_apps/baas_client.dart [hunk 4]
```diff
@@ -584,8 +584,12 @@ class BaasClient {
     final mongoServiceId = service['_id'] as String;
     final dynamic configDocs = await _get('groups/$_groupId/apps/$appId/services/$mongoServiceId/config');
     final dynamic flexibleSync = configDocs['flexible_sync'];
+    final dynamic clusterName = configDocs['clusterName'];
     flexibleSync["is_recovery_mode_disabled"] = !enable;
-    String data = jsonEncode(<String, dynamic>{'clusterName': configDocs['clusterName'], 'flexible_sync': flexibleSync});
+    String data = jsonEncode(<String, dynamic>{
+      if (clusterName != null) 'clusterName': configDocs['clusterName'],
```
Comment (nirinchev) [inline lib/src/cli/atlas_apps/baas_client.dart:590]: Why not use `clusterName` here?

Comment (nielsenko) [reply lib/src/cli/atlas_apps/baas_client.dart:590]: Oversight

**Pull request merged**